lib/string.js

```diff
@@ -70,6 +70,10 @@ internals.String.prototype._base = function (value, state, options) {
         if (this._flags.trim) {
             value = value.trim();
         }
+
+        if (this._flags.transform) {
```

Comment:
So, as per #586, I've implemented the `replace(...)` as a function simply transforming the value.

The `transform(value)` function returns the new value, now it does a RegExp replace, but it could do, really, anything...

It would be possible to move the function into `any.js` around line 580, where `_base` is called (if set), and thus allow for extensibility of transformation as requested in #608 ...

In `any.js` we should allow for a number of said functions to execute (one after another, maybe treating the `_flags.transform` as an array of functions).

Dunno, I precisely named the flag as `transform` to see whether this would be an idea you guys are open to, or how you think to implement it.

After looking @ the code for a bit, I _could_ try to do a quick implementation, but I honestly think that would need to be a different pull request.

Thoughts?